src/liballoc/btree/map.rs

```diff
@@ -213,7 +213,16 @@ impl<K: Clone, V: Clone> Clone for BTreeMap<K, V> {
             }
         }
 
-        clone_subtree(self.root.as_ref())
+        if self.len() == 0 {
+            // XXX: ideally we'd call `BTreeMap::new` here, but that has the
```

Comment:
Tidy is complaining XXX is deprecated notation; i would just drop the XXX, comment is fine without it